server/events/vcs/github_client.go

```diff
@@ -509,19 +509,27 @@ func (g *GithubClient) UpdateChecksStatus(ctx context.Context, request types.Upd
 	}
 
 	status, conclusion := g.resolveChecksStatus(request.State)
-	checkRunOutput := github.CheckRunOutput{
-		Title: &request.StatusName,
-	}
-	if request.Output != "" {
-		checkRunOutput.Text = &output
-	}
 
 	// Update checkrun if it exists and if it's not a rerun
 	// request.state is pending only when an operation starts. So, if the checkrun exists and the state is pending, it is a rerun.
 	if checkRun := g.findCheckRun(request.StatusName, checkRuns); checkRun != nil && request.State != models.PendingCommitStatus {
```

Comment:
can we reduce the nesting here?
https://docs.lyft.net/eng/atlantis/internal/team_practices/style_guide.html#reduce-nesting

Also this function is getting big, consider splitting into multiple functions for readability.

Reply:
Refactored it into smaller functions